src/java/org/apache/cassandra/repair/RepairJobDesc.java

```diff
@@ -61,6 +68,17 @@ public RepairJobDesc(UUID parentSessionId, UUID sessionId, String keyspace, Stri
         this.ranges = ranges;
     }
 
+    public UUID determanisticId()
```

Comment:
is it used to identify a repair job?  `sessionId` should be sufficient

Reply:
session can have multiple jobs, so its just a way to make sure each job is distinct; this is mostly only used for the vtables and not used in normal repair as the RepairJobDesc equality is what is used.

Reply:
its alway a way to make sure the different "tasks" (not a word we use in repair code) that a job does share the same ID (without changing serialization); the validation task for the repair job shares the id of the job for easy lookup

Reply:
Alternatively, we can concate `sessionId` with `keyspace_name` and `cf_name`, they should uniquely identify a `RepairJob` and easier to reason..

Reply:
so drop `parentSessionId` and `ranges`?  I can see `parentSessionId` as the session should be unique, but would prefer to keep `ranges` in case we can (even in the future) generate multiple jobs for the same table but cross different ranges (there is a ask for this on our side when a primary range is too large, we should be smart enough to split the repair, which would cause the same `sessionId, keyspace_name, cf_name` to conflict)